mlir/lib/Conversion/MIGraphXToTosa/MIGraphXToTosa.cpp

```diff
@@ -583,8 +583,9 @@ BroadcastConverter::matchAndRewrite(migraphx::BroadcastOp op, OpAdaptor adaptor,
   // We create a dummy multiplication with one with implicit broadcasting
   // because tosa does not have an explicit broadcast op
   auto oneTensor = rock::tosa::getOneTensor(rewriter, loc, outType);
-  auto mulWithOne = rock::tosa::getMulOp(rewriter, loc, sameRankReshapedOp,
-                                         oneTensor, elemType);
+  auto mulWithOne =
+      rock::tosa::getMulOp(rewriter, loc, sameRankReshapedOp, oneTensor,
+                           getTypeConverter()->convertType(elemType));
```

Comment:
nit; `newOutElementTy` is already defined earlier, you can use it directly
```suggestion
                           newOutElementTy);
```